File: challenge_9/jordan.java
// javac jordan.java
// java jordan "key" "plainText"

public class jordan {
	public static void main(String[] args) {
		System.out.println(encrypt(args[0], args[1]));
	}

	static String encrypt(String key, String plainText) {
		String encrypted = "";

		for (int textIndex = 0, keyIndex = 0; textIndex < plainText.length(); textIndex++) {
			char currentChar = plainText.charAt(textIndex);

			if (currentChar == ' ') {
				encrypted += ' ';
				keyIndex = ++keyIndex % key.length();
				continue;
			}

			encrypted += (char)((currentChar + key.charAt(keyIndex) - 2 * 'A') % 26 + 'A');
			keyIndex = ++keyIndex % key.length();
		}

		return encrypted;
	}
}

File: challenge_9/jordan.py
# python jordan.py "key" "plainText"

import sys

key = sys.argv[1]
text = sys.argv[2]

encrypted = ""
key_index = 0

for i in range(0, len(text)):
	if text[i] == ' ':
		encrypted += ' '
		key_index+=1
		key_index = key_index % len(key)
		continue

	encrypted += chr(((ord(text[i]) + ord(key[key_index]) - 2 * ord('A')) % 26 + ord('A')))
	key_index+=1
	key_index = key_index % len(key)

print encrypted